src/cli/cli.cpp

```diff
@@ -6698,6 +6698,37 @@ template <> otError Interpreter::Process<Cmd("txpower")>(Arg aArgs[])
     return error;
 }
 
+/**
+ * @cli targetpower (set)
+ * @code
+ * targetpower 12 2000
+ * Done
+ * @endcode
+ * @endcode
+ * @cparam targetpower @ca{channel} @ca{targetpower}
+ * @par
+ * Sets the target power in the unit of 0.01 dBm.
+ * @sa otPlatRadioSetChannelTargetPower
+ */
+template <> otError Interpreter::Process<Cmd("targetpower")>(Arg aArgs[])
+{
+    otError  error = OT_ERROR_NONE;
+    uint8_t  channel;
+    int16_t  targetPower;
+    uint32_t channelMask;
+
+    VerifyOrExit(!aArgs[0].IsEmpty() && !aArgs[1].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
```

Comment:
Since we call `aArgs[0].ParseAsUint8()` and `aArgs[1].ParseAsInt16()`, we can consider removing the check for `!IsEmpty()` here, since these calls will already check and ensure that `aArgs` is not empty (if it is, the `ParseAs{}` will return `OT_ERROR_INVALID_ARGS`).